pkg/driver/driver.go

```diff
@@ -137,6 +142,12 @@ func (d *BaseDriver) ListSnapshots(_ context.Context) (string, error) {
 	return "", fmt.Errorf("unimplemented")
 }
 
+func (d *BaseDriver) ForwardGuestAgent() bool {
```

Comment:
My concern is still here. 

My point is we don't need this. Instead of treating unix sock as fallback am saying lets treat it as a way drivers can use. But drivers need to decide on the communication medium. 

For eg: vz will always work on vsock, virtbox will always work on unix sock (This individual drivers should decide). Hostagent should not have that code

Reply:
I wanted to **keep** the hostagent tunneling as a fallback. It was handy when experimenting with external VMs.

For the VBox driver, it can probably use a serial console. Will do a rebase of it someday, it is not important...

Reply:
When we say fallback i wanted this to be a universal one not just for un configured driver.

Something like this  https://github.com/lima-vm/lima/commit/4640684a0e55ce0b72afce87133343ffbe3a7830. This is just a idea how we can do, based on the error we can fallback to this universal unix sock

Reply:
As long as it is done in the hostagent, and not copy/paste (or explicit) in the driver, that would work as well

Reply:
Not sure where to cancel the forwarding process, but maybe it would just die when the instance does?